### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [1]:
# Dependencies and Setup
import pandas as pd
import numpy as np
import pprint

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data_df = pd.read_csv(file_to_load)

# Starter Notebook Information

![starter_2_purchase_data](Images/starter_purchase_data.jpg)

In [2]:
# Show the purchase_data_df data
purchase_data_df

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

## Player Count

### Starter Notebook Information

![starter__player_count](Images/starter_player_count.jpg)

In [3]:
# Get the length of the series created by doing a value_count on it
total_players = len(purchase_data_df["SN"].value_counts())
total_players

576

In [4]:
# Create a DataFrame to store the result and update with total_players
total_player_count = pd.DataFrame({"Total Players":[total_players]})

<div class="alert alert-block alert-info">
    <b><h2><strong>Player Count result</strong></h2></b>
</div>

In [5]:
# Show the total player count
total_player_count

Total Players
0            576

## Purchasing Analysis (Total)

### Starter Notebook Information

![starter__purchasing_analysis_total](Images/starter_purchasing_analysis_total.jpg)

In [6]:
# Calcualte the number of unique items
unique_items = len(purchase_data_df['Item ID'].value_counts())
unique_items

179

In [7]:
# Calculate the average price
average_price = purchase_data_df['Price'].sum() / purchase_data_df['Purchase ID'].count()
average_price

3.0509871794871795

In [8]:
# Calculate the total number of purchases
number_purchases = purchase_data_df['Purchase ID'].count()
number_purchases

780

In [9]:
# Calculate total revenue for all purchases
total_revenue = purchase_data_df['Price'].sum()
total_revenue

2379.77

In [10]:
# Create a dataframe with all of our summary information
purchasing_totals_df = pd.DataFrame({"Number of Unique Items":[unique_items],
                                    "Average Price":[average_price],
                                    "Number of Purchases":[number_purchases],
                                    "Total Revenue":[total_revenue]  
                                    },)
purchasing_totals_df

Number of Unique Items  Average Price  Number of Purchases  Total Revenue
0                     179       3.050987                  780        2379.77

In [11]:
# Add formatting for the currancy data in 'Average Price' and 'Total Revenue'
purchasing_totals_df["Average Price"] = purchasing_totals_df["Average Price"].map("${:,.2f}".format)
purchasing_totals_df["Total Revenue"] = purchasing_totals_df["Total Revenue"].map("${:,.2f}".format)


<div class="alert alert-block alert-info">
    <b><h2><strong>Purchasing Analysis (Total) result</strong></h2></b>
</div>

In [12]:
# Show the Purchasing Analysis (Total) information
purchasing_totals_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780     $2,379.77

## Gender Demographics

### Starter Notebook Information

![starter_gender_demographics](Images/starter_gender_demographics.jpg)

In [13]:
# Group purchase_data by Gender
grouped_gender_df = purchase_data_df.groupby("Gender")

In [14]:
# Count the total of screen names "SN" by gender
total_sn_by_gender = grouped_gender_df.nunique()["SN"]
total_sn_by_gender

Gender
Female                    81
Male                     484
Other / Non-Disclosed     11
Name: SN, dtype: int64

In [15]:
# Calculate the percentage of players for each gender
percentage_of_players = total_sn_by_gender / total_players
percentage_of_players

Gender
Female                   0.140625
Male                     0.840278
Other / Non-Disclosed    0.019097
Name: SN, dtype: float64

In [16]:
# Create a dataframe to store our totals
gender_percent_summary_df = pd.DataFrame({"Total Count": total_sn_by_gender,"Percentage of Players": percentage_of_players})
gender_percent_summary_df

Total Count  Percentage of Players
Gender                                                   
Female                          81               0.140625
Male                           484               0.840278
Other / Non-Disclosed           11               0.019097

In [17]:
# Sort the data by Total Count
sorted_gender_percent_summary_df = gender_percent_summary_df.sort_values('Total Count',ascending=False)
sorted_gender_percent_summary_df

Total Count  Percentage of Players
Gender                                                   
Male                           484               0.840278
Female                          81               0.140625
Other / Non-Disclosed           11               0.019097

In [18]:
# Add formatting to the Percentage of Players data
sorted_gender_percent_summary_df["Percentage of Players"] = sorted_gender_percent_summary_df["Percentage of Players"].map("{:.2%}".format)


<div class="alert alert-block alert-info">
    <b><h2><strong>Gender Demographics results</strong></h2></b>
</div>

In [19]:
# Show the Gender Demographics information
sorted_gender_percent_summary_df

Total Count Percentage of Players
Gender                                                  
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

## Purchasing Analysis (Gender)

### Starter Notebook Information

![starter_purchasing_analysis_gender](Images/starter_purchasing_analysis_gender.jpg)

In [20]:
# Calculate the number of purchases per Gender
gender_number_purchases = grouped_gender_df['Purchase ID'].count()
gender_number_purchases

Gender
Female                   113
Male                     652
Other / Non-Disclosed     15
Name: Purchase ID, dtype: int64

In [21]:
# Calculate the average price per Gender
gender_average_price = grouped_gender_df['Price'].sum() / grouped_gender_df['Purchase ID'].count()
gender_average_price

Gender
Female                   3.203009
Male                     3.017853
Other / Non-Disclosed    3.346000
dtype: float64

In [22]:
# Calculate the total revenue per Gender
gender_total_revenue = grouped_gender_df['Price'].sum()
gender_total_revenue

Gender
Female                    361.94
Male                     1967.64
Other / Non-Disclosed      50.19
Name: Price, dtype: float64

In [23]:
# Group purchase_data by SN and by Gender in order to calculate the average purchase per person
grouped_sn_gender_df = purchase_data_df.groupby(["SN","Gender"])
grouped_sn_gender_df.head()

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

In [24]:
# Calculate the total purchase per SN and Gender
total_purchase_gender_by_sn = grouped_sn_gender_df['Price'].sum()
total_purchase_gender_by_sn

SN             Gender
Adairialis76   Male      2.28
Adastirin33    Female    4.48
Aeda94         Male      4.91
Aela59         Male      4.32
Aelaria33      Male      1.79
                         ... 
Yathecal82     Female    6.22
Yathedeu43     Male      6.02
Yoishirrala98  Female    4.58
Zhisrisu83     Male      7.89
Zontibe81      Male      8.03
Name: Price, Length: 576, dtype: float64

In [25]:
# Now group by Gender calculate the averge purchase per person by Gender
average_total_purchase_by_gender = total_purchase_gender_by_sn.groupby('Gender').mean()
average_total_purchase_by_gender

Gender
Female                   4.468395
Male                     4.065372
Other / Non-Disclosed    4.562727
Name: Price, dtype: float64

In [26]:
# Add the data into the summary DataFram3e
purchasing_analysis_gender_df = pd.DataFrame({"Purchase Count": gender_number_purchases,
                                              "Average Purchase Price": gender_average_price,
                                              "Total Purchase Value": gender_total_revenue,
                                             "Average Total Purchase Per Person": average_total_purchase_by_gender
                                            })
purchasing_analysis_gender_df

Purchase Count  Average Purchase Price  \
Gender                                                          
Female                            113                3.203009   
Male                              652                3.017853   
Other / Non-Disclosed              15                3.346000   

                       Total Purchase Value  Average Total Purchase Per Person  
Gender                                                                          
Female                               361.94                           4.468395  
Male                                1967.64                           4.065372  
Other / Non-Disclosed                 50.19                           4.562727

In [27]:
# Add formatting to the 3 currency columns of data
purchasing_analysis_gender_df["Average Purchase Price"] = purchasing_analysis_gender_df["Average Purchase Price"].map("${:,.2f}".format)
purchasing_analysis_gender_df["Total Purchase Value"] = purchasing_analysis_gender_df["Total Purchase Value"].map("${:,.2f}".format)
purchasing_analysis_gender_df["Average Total Purchase Per Person"] = purchasing_analysis_gender_df["Average Total Purchase Per Person"].map("${:,.2f}".format)


<div class="alert alert-block alert-info">
    <b><h2><strong>Purchasing Analysis (Gender) result</strong></h2></b>
</div>

In [28]:
# Show the Purchasing Analysis (Gender) information
purchasing_analysis_gender_df

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Average Total Purchase Per Person  
Gender                                                                        
Female                             $361.94                             $4.47  
Male                             $1,967.64                             $4.07  
Other / Non-Disclosed               $50.19                             $4.56

## Age Demographics

### Starter notebook information

![starter_age_demographics](Images/starter_age_demographics.jpg)

In [29]:
# Create bins in which to place values based upon age
bins = [0, 9.9, 13.9, 18.9, 23.9, 28.9, 33.9, 38.9, 125]

# Create labels for these bins
group_labels = ["  <10", "10-14", "15-19", "20-24", "25-29", "30-34",
                "35_39", "  40+"]

In [30]:
# Slice the data and place it into bins
purchase_data_df['Age Ranges'] = pd.cut(purchase_data_df["Age"],bins,labels=group_labels,include_lowest=True)
purchase_data_df

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price Age Ranges  
0    Extraction, Quickblade Of Trembling Hands   3.53      20-24  
1                            Frenzied Scimitar   1.56        40+  
2                                 Final Critic   4.88      25-29  
3                                  Blindscythe   3.27      25-29  
4                                         Fury   1.44      20-24  
..                                         ...    ...        ...  
775                                       Wolf   3.54      20-24  
776                           Exiled Doomblade   1.63      20-24  
777      Celeste, Incarnation of the Corrupted   3.46      20-24  
778                               Final Critic   4.19        <10  
779                                       Dawn   4.60      25-29  

[780 rows x 8 columns]

In [31]:
# Group the data by 'Age Ranges'
grouped_age_ranges_df = purchase_data_df.groupby('Age Ranges')

In [32]:
# Count the total of screen names "SN" by age group
total_sn_by_age_ranges = grouped_age_ranges_df["SN"].nunique()
total_sn_by_age_ranges

Age Ranges
  <10     17
10-14     20
15-19     92
20-24    227
25-29    115
30-34     55
35_39     32
  40+     18
Name: SN, dtype: int64

In [33]:
# # Total count by gender and divivde by total players 
percentage_of_players_by_age_ranges = total_sn_by_age_ranges / total_players
percentage_of_players_by_age_ranges

Age Ranges
  <10    0.029514
10-14    0.034722
15-19    0.159722
20-24    0.394097
25-29    0.199653
30-34    0.095486
35_39    0.055556
  40+    0.031250
Name: SN, dtype: float64

In [34]:
# Add the data into the summary DataFrame
age_range_percent_summary_df = pd.DataFrame({"Total Count": total_sn_by_age_ranges,"Percentage of Players": percentage_of_players_by_age_ranges})
age_range_percent_summary_df

Total Count  Percentage of Players
Age Ranges                                    
  <10                17               0.029514
10-14                20               0.034722
15-19                92               0.159722
20-24               227               0.394097
25-29               115               0.199653
30-34                55               0.095486
35_39                32               0.055556
  40+                18               0.031250

In [35]:
# Add percentage formatting to the 'Percentage of Players' data
age_range_percent_summary_df["Percentage of Players"] = age_range_percent_summary_df["Percentage of Players"].map("{:.2%}".format)

<div class="alert alert-block alert-info">
    <b><h2><strong>Age Demographics results</strong></h2></b>
</div>

In [36]:
# Show the Age Demographics information
age_range_percent_summary_df

Total Count Percentage of Players
Age Ranges                                   
  <10                17                 2.95%
10-14                20                 3.47%
15-19                92                15.97%
20-24               227                39.41%
25-29               115                19.97%
30-34                55                 9.55%
35_39                32                 5.56%
  40+                18                 3.12%

## Purchasing Analysis (Age)

### Starter notebook information

![starter_purchasing_analysis_age](Images/starter_purchasing_analysis_age.jpg)

In [37]:
# Calculate the number of purchase per 'Age Range'
age_ranges_number_purchases = grouped_age_ranges_df['Purchase ID'].count()
age_ranges_number_purchases

Age Ranges
  <10     23
10-14     26
15-19    115
20-24    321
25-29    155
30-34     77
35_39     44
  40+     19
Name: Purchase ID, dtype: int64

In [38]:
# Calculate the average price per 'Age Range'
age_ranges_average_price = grouped_age_ranges_df['Price'].sum() / grouped_age_ranges_df['Purchase ID'].count()
age_ranges_average_price

Age Ranges
  <10    3.353478
10-14    2.918077
15-19    3.041913
20-24    3.033707
25-29    3.019290
30-34    2.949351
35_39    3.329091
  40+    3.240000
dtype: float64

In [39]:
# Calculate the total revenue per 'Age Range'
age_ranges_total_revenue = grouped_age_ranges_df['Price'].sum()
age_ranges_total_revenue

Age Ranges
  <10     77.13
10-14     75.87
15-19    349.82
20-24    973.82
25-29    467.99
30-34    227.10
35_39    146.48
  40+     61.56
Name: Price, dtype: float64

In [40]:
# Group purchase_data by Gender
grouped_sn_age_ranges_df = purchase_data_df.groupby(["SN","Age Ranges"])
grouped_sn_age_ranges_df.head()

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price Age Ranges  
0    Extraction, Quickblade Of Trembling Hands   3.53      20-24  
1                            Frenzied Scimitar   1.56        40+  
2                                 Final Critic   4.88      25-29  
3                                  Blindscythe   3.27      25-29  
4                                         Fury   1.44      20-24  
..                                         ...    ...        ...  
775                                       Wolf   3.54      20-24  
776                           Exiled Doomblade   1.63      20-24  
777      Celeste, Incarnation of the Corrupted   3.46      20-24  
778                               Final Critic   4.19        <10  
779                                       Dawn   4.60      25-29  

[780 rows x 8 columns]

In [41]:
# Calculate the total purchases by 'Age Range'
total_purchase_age_ranges_by_sn = grouped_sn_age_ranges_df['Price'].sum()
total_purchase_age_ranges_by_sn

SN            Age Ranges
Adairialis76    <10          NaN
              10-14          NaN
              15-19         2.28
              20-24          NaN
              25-29          NaN
                            ... 
Zontibe81     20-24         8.03
              25-29          NaN
              30-34          NaN
              35_39          NaN
                40+          NaN
Name: Price, Length: 4608, dtype: float64

In [42]:
# Calculate the average total purchase for each age range
average_total_purchase_by_age_ranges = total_purchase_age_ranges_by_sn.groupby('Age Ranges').mean()
average_total_purchase_by_age_ranges

Age Ranges
  <10    4.537059
10-14    3.793500
15-19    3.802391
20-24    4.289956
25-29    4.069478
30-34    4.129091
35_39    4.577500
  40+    3.420000
Name: Price, dtype: float64

In [44]:
# Add the data into the summary DataFram3e
purchasing_analysis_age_ranges_df = pd.DataFrame({"Purchase Count": age_ranges_number_purchases,
                                              "Average Purchase Price": age_ranges_average_price,
                                              "Total Purchase Value": age_ranges_total_revenue,
                                             "Average Total Purchase Per Person": average_total_purchase_by_age_ranges
                                            })
purchasing_analysis_age_ranges_df

Purchase Count  Average Purchase Price  Total Purchase Value  \
Age Ranges                                                                 
  <10                   23                3.353478                 77.13   
10-14                   26                2.918077                 75.87   
15-19                  115                3.041913                349.82   
20-24                  321                3.033707                973.82   
25-29                  155                3.019290                467.99   
30-34                   77                2.949351                227.10   
35_39                   44                3.329091                146.48   
  40+                   19                3.240000                 61.56   

            Average Total Purchase Per Person  
Age Ranges                                     
  <10                                4.537059  
10-14                                3.793500  
15-19                                3.802391  
20-24                                4.289956  
25-29                                4.069478  
30-34                                4.129091  
35_39                                4.577500  
  40+                                3.420000

In [45]:
# Add formatting to the 3 currency fields of data
purchasing_analysis_age_ranges_df["Average Purchase Price"] = purchasing_analysis_age_ranges_df["Average Purchase Price"].map("${:,.2f}".format)
purchasing_analysis_age_ranges_df["Total Purchase Value"] = purchasing_analysis_age_ranges_df["Total Purchase Value"].map("${:,.2f}".format)
purchasing_analysis_age_ranges_df["Average Total Purchase Per Person"] = purchasing_analysis_age_ranges_df["Average Total Purchase Per Person"].map("${:,.2f}".format)


<div class="alert alert-block alert-info">
    <b><h2><strong>Purchasing Analysis (Age) result</strong></h2></b>
</div>

In [46]:
# Show the Purchasing Analysis (Age) information
purchasing_analysis_age_ranges_df

Purchase Count Average Purchase Price Total Purchase Value  \
Age Ranges                                                               
  <10                   23                  $3.35               $77.13   
10-14                   26                  $2.92               $75.87   
15-19                  115                  $3.04              $349.82   
20-24                  321                  $3.03              $973.82   
25-29                  155                  $3.02              $467.99   
30-34                   77                  $2.95              $227.10   
35_39                   44                  $3.33              $146.48   
  40+                   19                  $3.24               $61.56   

           Average Total Purchase Per Person  
Age Ranges                                    
  <10                                  $4.54  
10-14                                  $3.79  
15-19                                  $3.80  
20-24                                  $4.29  
25-29                                  $4.07  
30-34                                  $4.13  
35_39                                  $4.58  
  40+                                  $3.42

## Top Spenders

# Starter notebook information

![starter_top_spenders](Images/starter_top_spenders.jpg)

In [47]:
# Group the purchase data by screen name ('SN')
grouped_by_sn_df = purchase_data_df.groupby('SN')

In [48]:
# Calculate the number of purchase for each user
sn_number_purchases = grouped_by_sn_df['Purchase ID'].count()
sn_number_purchases

SN
Adairialis76     1
Adastirin33      1
Aeda94           1
Aela59           1
Aelaria33        1
                ..
Yathecal82       3
Yathedeu43       2
Yoishirrala98    1
Zhisrisu83       2
Zontibe81        3
Name: Purchase ID, Length: 576, dtype: int64

In [49]:
# Calculate how many users had made 1 - 5 purchases
sn_number_purchases.value_counts()

1    414
2    124
3     35
4      2
5      1
Name: Purchase ID, dtype: int64

In [50]:
# Calculate the average price paid per user
sn_average_price = grouped_by_sn_df['Price'].sum() / grouped_by_sn_df['Purchase ID'].count()
sn_average_price

SN
Adairialis76     2.280000
Adastirin33      4.480000
Aeda94           4.910000
Aela59           4.320000
Aelaria33        1.790000
                   ...   
Yathecal82       2.073333
Yathedeu43       3.010000
Yoishirrala98    4.580000
Zhisrisu83       3.945000
Zontibe81        2.676667
Length: 576, dtype: float64

In [51]:
# Calculate the total purchase amount made by each user
sn_total_revenue = grouped_by_sn_df['Price'].sum()
sn_total_revenue

SN
Adairialis76     2.28
Adastirin33      4.48
Aeda94           4.91
Aela59           4.32
Aelaria33        1.79
                 ... 
Yathecal82       6.22
Yathedeu43       6.02
Yoishirrala98    4.58
Zhisrisu83       7.89
Zontibe81        8.03
Name: Price, Length: 576, dtype: float64

In [52]:
# Add the data into the summary DataFram3e
top_spenders_df = pd.DataFrame({"Purchase Count": sn_number_purchases,
                                              "Average Purchase Price": sn_average_price,
                                              "Total Purchase Value": sn_total_revenue,
                                            })
top_spenders_df

Purchase Count  Average Purchase Price  Total Purchase Value
SN                                                                         
Adairialis76                1                2.280000                  2.28
Adastirin33                 1                4.480000                  4.48
Aeda94                      1                4.910000                  4.91
Aela59                      1                4.320000                  4.32
Aelaria33                   1                1.790000                  1.79
...                       ...                     ...                   ...
Yathecal82                  3                2.073333                  6.22
Yathedeu43                  2                3.010000                  6.02
Yoishirrala98               1                4.580000                  4.58
Zhisrisu83                  2                3.945000                  7.89
Zontibe81                   3                2.676667                  8.03

[576 rows x 3 columns]

In [53]:
# Sort the data by Total Count
sorted_top_speners_df = top_spenders_df.sort_values('Total Purchase Value',ascending=False)
sorted_top_speners_df.head(5)

Purchase Count  Average Purchase Price  Total Purchase Value
SN                                                                       
Lisosia93                 5                3.792000                 18.96
Idastidru52               4                3.862500                 15.45
Chamjask73                3                4.610000                 13.83
Iral74                    4                3.405000                 13.62
Iskadarya95               3                4.366667                 13.10

In [54]:
# Add formatting to the two currency columns
sorted_top_speners_df["Average Purchase Price"] = sorted_top_speners_df["Average Purchase Price"].map("${:,.2f}".format)
sorted_top_speners_df["Total Purchase Value"] = sorted_top_speners_df["Total Purchase Value"].map("${:,.2f}".format)


<div class="alert alert-block alert-info">
    <b><h2><strong>Top Spenders result</strong></h2></b>
</div>

In [55]:
# Show the Top Spenders
sorted_top_speners_df.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items

# Starter notebook information

![starter_most_popular_items](Images/starter_most_popular_items.jpg)

In [56]:
# Create a dataframe of the 'Item ID', 'Item Name', and 'Price'
items_df = purchase_data_df[['Item ID','Item Name','Price']]
items_df

Item ID                                  Item Name  Price
0        108  Extraction, Quickblade Of Trembling Hands   3.53
1        143                          Frenzied Scimitar   1.56
2         92                               Final Critic   4.88
3        100                                Blindscythe   3.27
4        131                                       Fury   1.44
..       ...                                        ...    ...
775       60                                       Wolf   3.54
776      164                           Exiled Doomblade   1.63
777       67      Celeste, Incarnation of the Corrupted   3.46
778       92                               Final Critic   4.19
779       50                                       Dawn   4.60

[780 rows x 3 columns]

In [57]:
# Group the data by 'Item ID' and 'Item Name'
grouped_items_df = items_df.groupby(['Item ID','Item Name'])
grouped_items_df.head()

Item ID                                  Item Name  Price
0        108  Extraction, Quickblade Of Trembling Hands   3.53
1        143                          Frenzied Scimitar   1.56
2         92                               Final Critic   4.88
3        100                                Blindscythe   3.27
4        131                                       Fury   1.44
..       ...                                        ...    ...
764      113                          Solitude's Reaver   4.07
765      130                                      Alpha   2.07
766       58          Freak's Bite, Favor of Holy Might   4.14
777       67      Celeste, Incarnation of the Corrupted   3.46
779       50                                       Dawn   4.60

[690 rows x 3 columns]

In [58]:
# Count the number of purchase for each item
item_number_purchases = grouped_items_df['Item ID'].count()
item_number_purchases

Item ID  Item Name                                   
0        Splinter                                         4
1        Crucifer                                         4
2        Verdict                                          6
3        Phantomlight                                     6
4        Bloodlord's Fetish                               5
                                                         ..
178      Oathbreaker, Last Hope of the Breaking Storm    12
179      Wolf, Promise of the Moonwalker                  6
181      Reaper's Toll                                    5
182      Toothpick                                        3
183      Dragon's Greatsword                              3
Name: Item ID, Length: 179, dtype: int64

In [59]:
# Calculate the total money paid for each item
item_total_revenue = grouped_items_df['Price'].sum()
item_total_revenue

Item ID  Item Name                                   
0        Splinter                                         5.12
1        Crucifer                                        11.77
2        Verdict                                         14.88
3        Phantomlight                                    14.94
4        Bloodlord's Fetish                               8.50
                                                         ...  
178      Oathbreaker, Last Hope of the Breaking Storm    50.76
179      Wolf, Promise of the Moonwalker                 26.88
181      Reaper's Toll                                    8.30
182      Toothpick                                       12.09
183      Dragon's Greatsword                              3.27
Name: Price, Length: 179, dtype: float64

In [60]:
# Calculate the item price
item_price = item_total_revenue / item_number_purchases
item_price

Item ID  Item Name                                   
0        Splinter                                        1.2800
1        Crucifer                                        2.9425
2        Verdict                                         2.4800
3        Phantomlight                                    2.4900
4        Bloodlord's Fetish                              1.7000
                                                          ...  
178      Oathbreaker, Last Hope of the Breaking Storm    4.2300
179      Wolf, Promise of the Moonwalker                 4.4800
181      Reaper's Toll                                   1.6600
182      Toothpick                                       4.0300
183      Dragon's Greatsword                             1.0900
Length: 179, dtype: float64

In [61]:
# Add the data into the summary DataFrame for Most Popular Items
most_popular_df = pd.DataFrame({"Purchase Count": item_number_purchases,
                                "Item Price": item_price,
                                "Total Purchase Value": item_total_revenue,
                              })
most_popular_df

Purchase Count  \
Item ID Item Name                                                      
0       Splinter                                                   4   
1       Crucifer                                                   4   
2       Verdict                                                    6   
3       Phantomlight                                               6   
4       Bloodlord's Fetish                                         5   
...                                                              ...   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
179     Wolf, Promise of the Moonwalker                            6   
181     Reaper's Toll                                              5   
182     Toothpick                                                  3   
183     Dragon's Greatsword                                        3   

                                                      Item Price  \
Item ID Item Name                                                  
0       Splinter                                          1.2800   
1       Crucifer                                          2.9425   
2       Verdict                                           2.4800   
3       Phantomlight                                      2.4900   
4       Bloodlord's Fetish                                1.7000   
...                                                          ...   
178     Oathbreaker, Last Hope of the Breaking Storm      4.2300   
179     Wolf, Promise of the Moonwalker                   4.4800   
181     Reaper's Toll                                     1.6600   
182     Toothpick                                         4.0300   
183     Dragon's Greatsword                               1.0900   

                                                      Total Purchase Value  
Item ID Item Name                                                           
0       Splinter                                                      5.12  
1       Crucifer                                                     11.77  
2       Verdict                                                      14.88  
3       Phantomlight                                                 14.94  
4       Bloodlord's Fetish                                            8.50  
...                                                                    ...  
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
179     Wolf, Promise of the Moonwalker                              26.88  
181     Reaper's Toll                                                 8.30  
182     Toothpick                                                    12.09  
183     Dragon's Greatsword                                           3.27  

[179 rows x 3 columns]

In [62]:
# Sort the data by Total Count
sorted_most_popular_df = most_popular_df.sort_values('Purchase Count',ascending=False)
sorted_most_popular_df.head(5)

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                      Item Price  \
Item ID Item Name                                                  
92      Final Critic                                    4.614615   
178     Oathbreaker, Last Hope of the Breaking Storm    4.230000   
145     Fiery Glass Crusader                            4.580000   
132     Persuasion                                      3.221111   
108     Extraction, Quickblade Of Trembling Hands       3.530000   

                                                      Total Purchase Value  
Item ID Item Name                                                           
92      Final Critic                                                 59.99  
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
145     Fiery Glass Crusader                                         41.22  
132     Persuasion                                                   28.99  
108     Extraction, Quickblade Of Trembling Hands                    31.77

In [63]:
# Add formatting for the two currency columns
sorted_most_popular_df["Item Price"] = sorted_most_popular_df["Item Price"].map("${:,.2f}".format)
sorted_most_popular_df["Total Purchase Value"] = sorted_most_popular_df["Total Purchase Value"].map("${:,.2f}".format)


<div class="alert alert-block alert-info">
    <b><h2><strong>Most Popular Items result</strong></h2></b>
</div>

In [64]:
# Show the Five Most Popular Items
sorted_most_popular_df.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items

# Starter notebook information

![starter_most_profitable_items](Images/starter_most_profitable_items.jpg)

In [65]:
# Sort the data by Total Count
sorted_most_profitable_df = most_popular_df.sort_values('Total Purchase Value',ascending=False)
sorted_most_profitable_df.head(5)

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                      Item Price  \
Item ID Item Name                                                  
92      Final Critic                                    4.614615   
178     Oathbreaker, Last Hope of the Breaking Storm    4.230000   
82      Nirvana                                         4.900000   
145     Fiery Glass Crusader                            4.580000   
103     Singed Scalpel                                  4.350000   

                                                      Total Purchase Value  
Item ID Item Name                                                           
92      Final Critic                                                 59.99  
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
82      Nirvana                                                      44.10  
145     Fiery Glass Crusader                                         41.22  
103     Singed Scalpel                                               34.80

In [66]:
# Add formatting to our table for the Item Price and Total Purchase Value
sorted_most_profitable_df["Item Price"] = sorted_most_profitable_df["Item Price"].map("${:,.2f}".format)
sorted_most_profitable_df["Total Purchase Value"] = sorted_most_profitable_df["Total Purchase Value"].map("${:,.2f}".format)


<div class="alert alert-block alert-info">
    <b><h2><strong>Most Profitable Items result</strong></h2></b>
</div>

In [67]:
# Show the Five Most Profitable Items
sorted_most_profitable_df.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80


<div class="alert alert-block alert-info">
    <b><h1><strong>Observable Trends</strong></h1></b>
</div>

The players that purchase in game items tend to be male.  There are  **84.03%** of players purchasing in game items that are male.  This is in contrast with only **14.06%** in game purchasers being female.  Any future marketing should target males.


The age range of **20-24** outspent all other groups with 321 purchases spending a total of 973.82 dollars.  The next age range is **25-29** which purchased 155 for a total of 467.99 dollars.  And the thrid is **15-19** year olds who bought 115 items for 349.82.   These three age ranges combined account for 591 out of 780 purchase and 1791.63 of 2379.77 dollars spent.  Since this represents 75.8% of purchases and 74.7% of the dollars spent, it would make sense for any future marketing to target **15-29** years old.

**Final Critic**, **OathBreaker, Last Hope of the Breaking Storm**, and **Fiery Glass Crusider** are all in the top 5 of popular and profitable items.   It would make sense to target these three products in any future marketing.

Very few of the users make more than one in game purchase. Future marketing could target those that have made one purchase to see if they could be enticed to make additional purchases.

| # of Purchases |  # of Users |
|----------------|-------------|
|               1|          414|
|               2|          124|
|               3|           35|
|               4|            2|
|               5|            1|